app/Console/Commands/CreateMasterVoucherLogReport.php

```diff
@@ -0,0 +1,308 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Carbon\Carbon;
+use DB;
+use Excel;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Console\Command;
+use Log;
+use Maatwebsite\Excel\Writers\LaravelExcelWriter;
+use ZipArchive;
+
+class CreateMasterVoucherLogReport extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'arc:createMVLReport 
+                            {--force : Execute without confirmation, eg for automation}
+                            {--no-zip : Don\'t wrap files in a single archive}
+                            {--plain : Don\'t Encrypt contents of files}
+                            '
+    ;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Creates, encrypts and stores the MVL report under in /storage';
+
+    /**
+     * The default disk we want.
+     *
+     * @var string $disk
+     */
+    private $disk = 'enc';
+
+    /**
+     * The default archive name.
+     *
+     * @var string $archiveName
+     */
+    private $archiveName = "MVLReport.zip";
+
+    /**
+     * The sheet headers.
+     *
+     * @var array $headers
+     */
+    private $headers = [
+        'Area',
+        'Voucher Code',
+        'Dispatch Date',
+        'Disbursed Date',
+        'RVID',
+        'Participant',
+        'Payment Request Date',
+        'Trader Name',
+        'Retail Outlet',
+        'Reimbursed Date'
+    ];
+
+    /**
+     * The report's query template
+     *
+     * TODO: refactor this as eloquent lookups when it's finalised.
```

Comment:
Will you do the refactor in this PR - or mean at a later date? You know I'm not keen on raw SQL. :smile:

Reply:
At a later date. The sheer complexity of this means that SQL is probably the most efficient way to pull this particular report. I'd like it to be refactored, as the functions might be useful in a future "admin" sprint.

Reply:
Yeah - it looks nuts. Agree that functions will likely turn out to be useful but I guess you are in a rush?

Reply:
time - money in this case.